test/fork/invariant/Invariants.t.sol

```diff
@@ -0,0 +1,191 @@
+                no.totalAddedKeys,
+                no.totalDepositedKeys,
+                "assert added >= deposited"
+            );
+            assertGe(
+                no.totalDepositedKeys,
+                no.totalWithdrawnKeys,
+                "assert deposited >= withdrawn"
+            );
+            assertGe(
+                no.totalVettedKeys,
+                no.totalDepositedKeys,
+                "assert vetted >= deposited"
+            );
+
+            assertGe(
+                no.totalDepositedKeys - no.totalExitedKeys,
+                no.stuckValidatorsCount,
+                "assert deposited - exited >= stuck"
+            );
+
+            assertGe(
+                no.totalAddedKeys,
+                no.depositableValidatorsCount + no.totalWithdrawnKeys,
+                "assert added >= depositable + withdrawn"
+            );
+            assertLe(
+                no.totalWithdrawnKeys,
+                no.totalExitedKeys,
+                "assert withdrawn <= exited"
+            );
+            assertLe(
+                no.depositableValidatorsCount,
+                no.totalAddedKeys - no.totalDepositedKeys,
+                "assert depositable <= added - deposited"
+            );
+
+            assertNotEq(
+                no.proposedManagerAddress,
+                no.managerAddress,
+                "assert proposed != manager"
+            );
+            assertNotEq(
+                no.proposedRewardAddress,
+                no.rewardAddress,
+                "assert proposed != reward"
+            );
+            assertNotEq(no.managerAddress, address(0), "assert manager != 0");
+            assertNotEq(no.rewardAddress, address(0), "assert reward != 0");
+
+            totalExitedValidators += no.totalExitedKeys;
+            totalDepositedValidators += no.totalDepositedKeys;
+            totalDepositableValidators += no.depositableValidatorsCount;
+        }
+
+        (
+            uint256 _totalExitedValidators,
+            uint256 _totalDepositedValidators,
+            uint256 _depositableValidatorsCount
+        ) = csm.getStakingModuleSummary();
+        assertTrue(totalExitedValidators <= _totalExitedValidators);
+        assertTrue(totalDepositedValidators <= _totalDepositedValidators);
+        assertTrue(totalDepositableValidators <= _depositableValidatorsCount);
+    }
+
+    mapping(uint256 => uint256) batchKeys;
+
+    function test_enqueuedCount() public {
+        uint256 noCount = csm.getNodeOperatorsCount();
+        NodeOperator memory no;
+
+        uint128 queueIndex;
+        (uint128 head, uint128 length) = csm.depositQueue();
+        for (uint128 i = 0; i < length; i++) {
+            queueIndex = head + i;
+            Batch item = csm.depositQueueItem(queueIndex);
+            batchKeys[item.noId()] += item.keys();
+            assertLe(
+                no.enqueuedCount,
+                item.keys(),
+                "assert enqueued <= batch keys"
+            );
```

Comment:
What if there are multiple batches?

Reply:
removed due to it duplicates the check below